boa_engine/src/object/mod.rs

```diff
@@ -758,6 +786,40 @@ impl Object {
         }
     }
 
+    /// Checks if it's a `Generator` object.
+    #[inline]
+    pub fn is_generator(&self) -> bool {
+        matches!(
+            self.data,
+            ObjectData {
+                kind: ObjectKind::Generator(_),
+                ..
+            }
+        )
+    }
+
+    #[inline]
+    pub fn as_generator(&self) -> Option<&Generator> {
+        match self.data {
+            ObjectData {
+                kind: ObjectKind::Generator(ref generator),
+                ..
+            } => Some(generator),
+            _ => None,
+        }
+    }
+
+    #[inline]
+    pub fn as_generator_mut(&mut self) -> Option<&mut Generator> {
+        match self.data {
+            ObjectData {
+                kind: ObjectKind::Generator(ref mut generator),
+                ..
+            } => Some(generator),
+            _ => None,
+        }
+    }
```

Comment:
Same to the previous function, would be useful to have a small text to explain what they do, even if it seems obvious (I think we are missing this in other functions too)